elm/wizard.py

```diff
@@ -441,31 +447,34 @@ def __init__(self, db_host, db_port, db_name,
             GPT.
         """
         boto3 = try_import('boto3')
-        psycopg2 = try_import('psycopg2')
+        self.psycopg2 = try_import('psycopg2')
 
-        self.db_schema = db_schema
-        self.db_table = db_table
         if meta_columns is None:
             self.meta_columns = ['title', 'url']
         else:
             self.meta_columns = meta_columns
 
         if cursor is None:
-            db_user = os.getenv("EWIZ_DB_USER")
-            db_password = os.getenv('EWIZ_DB_PASSWORD')
-            assert db_user is not None, "Must set EWIZ_DB_USER!"
-            assert db_password is not None, "Must set EWIZ_DB_PASSWORD!"
-            self.conn = psycopg2.connect(user=db_user,
-                                         password=db_password,
-                                         host=db_host,
-                                         port=db_port,
-                                         database=db_name)
+            self.db_host = db_host
+            self.db_port = db_port
+            self.db_name = db_name
+            self.db_schema = db_schema
+            self.db_table = db_table
+            self.db_user = os.getenv("EWIZ_DB_USER")
+            self.db_password = os.getenv('EWIZ_DB_PASSWORD')
+            assert self.db_user is not None, "Must set EWIZ_DB_USER!"
+            assert self.db_password is not None, "Must set EWIZ_DB_PASSWORD!"
+            self.db_kwargs = dict(user=self.db_user, password=self.db_password,
+                                  host=self.db_host, port=self.db_port,
+                                  database=self.db_name)
+            self.conn = self.psycopg2.connect(**self.db_kwargs)
 
             self.cursor = self.conn.cursor()
         else:
             self.cursor = cursor
 
         self.tag = tag
+        self.probes = probes
```

Comment:
I would love to get your input on this. Our apps use `query_vector_db` behind the scenes and `probes` is a value that is specific to each db, for example for the ewiz db we need 25 probes and for askOEDI we need 10. Since we aren't directly accessing `query_vector_db` I don't think we can modify the value without including it as an attribute. Does that make sense?

Reply:
Hmmm, I guess I would typically recommend exposing it to the top-most level function that _is_ used in external code (likely `chat(...)`?). However, you may be trying to follow the base signature for `chat`, so in that case what you have here is probably ok. I would say leave it as is until a better use-pattern shows up in the future :)

Reply:
Sounds good, thank you!